contracts/wawaka/memory-test/memory-test.cpp

```diff
@@ -0,0 +1,134 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <malloc.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "Dispatch.h"
+
+#include "KeyValue.h"
+#include "Environment.h"
+#include "Message.h"
+#include "Response.h"
+#include "StringArray.h"
+#include "Value.h"
+#include "WasmExtensions.h"
+
+static KeyValueStore meta_store("meta");
+
+const StringArray owner_key("owner");
+const StringArray default_key("key");
+const StringArray default_val("_");
+
+// -----------------------------------------------------------------
+// NAME: initialize_contract
+// -----------------------------------------------------------------
+bool initialize_contract(const Environment& env, Response& rsp)
+{
+    // save owner information
+    const StringArray owner_val(env.creator_id_);
+
+    if (! meta_store.set(owner_key, owner_val))
+        return rsp.error("failed to save creator metadata");
+
+    return rsp.success(true);
+}
+
+// -----------------------------------------------------------------
+// NAME: many_keys_test
+// -----------------------------------------------------------------
+bool many_keys_test(const Message& msg, const Environment& env, Response& rsp)
+{
+    int num_keys((int)msg.get_number("num_keys"));
+
+    int i = 0;
+    for (i = 0; i < num_keys; i++) {
+        StringArray key(12);
+        sprintf((char *)key.value_, "%d", i);
+        if (!meta_store.set(key, default_val))
+            return rsp.error("failed to store value");
+    }
+
+    ww::value::Number v((double)i);
+    return rsp.value(v, true);
+}
+
+// -----------------------------------------------------------------
+// NAME: big_key_test
+// -----------------------------------------------------------------
+bool big_key_test(const Message& msg, const Environment& env, Response& rsp)
+{
+    int num_chars((int)msg.get_number("num_chars"));
+
+    StringArray key(num_chars);
+    for (int i = 0; i < num_chars; i++) {
+        key.set('a', i);
+    }
+
+    if (!meta_store.set(key, default_val))
+        return rsp.error("failed to store value");
+
+    ww::value::Number v((double)key.size());
+    return rsp.value(v, true);
+}
+
+// -----------------------------------------------------------------
+// NAME: big_value_test
+// -----------------------------------------------------------------
+bool big_value_test(const Message& msg, const Environment& env, Response& rsp)
+{
+    int num_chars((int)msg.get_number("num_chars"));
+
+    StringArray value(num_chars);
+    for (int i = 0; i < num_chars; i++) {
+        value.set((char)*default_val.value_, i);
+    }
+
+    if (!meta_store.set(default_key, value))
+        return rsp.error("failed to store value");
+
+    ww::value::Number v((double)value.size());
+    return rsp.value(v, true);
+}
+
```

Comment:
just curious as to why there isn't an all-in-one test "many-keys, all with bigs keys and values", if the goal is stress testing?

Reply:
This would be interesting. I can add in another function that does this.